OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -76,7 +79,16 @@ void ModelComponent::extendConnect(Component &root)
         connectToModel(*model);
 }
 
-
+void ModelComponent::extendFinalizeFromProperties()
+{
+    Super::extendFinalizeFromProperties();
+    int geomSize = getProperty_GeometrySet().size();
+    if (geomSize > 0){
+        for (int i = 0; i < geomSize; ++i){
+            addComponent(&upd_GeometrySet(i));
```

Comment:
Sorry @sherm1 I thought you meant ModelComponent's constructor. I don't particularly like that Model construction is calling `finalizeFromProperties` either, but not because it is necessarily dangerous. We want to minimize calls to finalizeFromProperties because it could be loading big data files or doing some computation/initialization of modeling parameters based on property values, so we don't want deserialization or default construction to invoke it. The faux pas that should not be made is to call `addComponent()` outside of `finalizeFromProperties`.